### DISCUSSION TITLE
What are the plans regarding recursion?

### DISCUSSION THREAD

**Discussion by wrq** [fubark/cyber#45] [Ideas]
Will cyber support TCO or automatic trampolining for writing recursive functions neatly?

Thanks!

**Comment by fubark** [comment]
Yes, I want TCO to be implemented and enabled by default (well only if it doesn't affect compile time perf too much). If people want to debug those functions, we would enable a flag to skip TCO.

**Reply by wrq** [reply]
Oh, this is wonderful news. As time allows, I plan on putting some very small demos together with Cyber, I think you're building something really special here.

Thanks for the prompt response as well!

**Reply by fubark** [reply]
Can't wait to see it!
